meta/classes/base.bbclass

```diff
@@ -77,9 +77,8 @@ base_update_conf_value() {
     local key=$2
     local value=$3
     local key_prefix=$4
-    local exists=`grep "^[^\#]*${key}.*=" "${config_file}"`
 
-    if [ -z ${exists} ] ; then
```

Comment:
`[ -z "${exists}" ]` will work for you, if you need `exists` later.

Reply:
then anyways it will need to be changed so it can handle more complicated use-cases

Reply:
done